micro-rdk-installer/src/main.rs

```diff
@@ -214,13 +239,18 @@ fn write_credentials_to_app_binary(
     Ok(())
 }
 
-fn flash(args: WriteFlashArgs, config: &Config, app_path: PathBuf) -> Result<(), Error> {
+fn flash(
+    flash_args: FlashArgs,
+    monitor_args: MonitorArgs,
```

Comment:
Why do we pass MonitorArgs now? what's the upside? how does the new invocation looks like?

Reply:
many of `espflash`'s method's, such as `Flasher::connect`, take these structs as parameters. Sadly, because the structs are all annotated as [`non-exhaustive`](https://github.com/esp-rs/espflash/blob/890f17ac46ea8d579118ae5a726c3a940ede9065/espflash/src/cli/mod.rs#L53), they can't just be initialized directly within our functions (ex `let m_args = MonitorArgs{..}`). 

This means that the only way we can create these input arguments via clap.
`cargo-espflash` highlights this in that it also passes all there structs directly or with a thin wrapper.
In addition, we have to be mindful about what `espflash` Commands we wrap in our own because the flattened fields of all inner commands must be unique, which sadly isn't always the case. `espflash` as a library isn't very ergonomic.

The new invocation is `cargo r -p micro-rdk-installer -- update-app-image --bootloader target/xtensa-esp32-espidf/esp32-server.bin`, with the breaking change being the `--bootloader` flag to specify the binary. Bumping `espflash` to 3.0 was a breaking change so it seemed an opportune moment to also have breaking change for the installer.

Reply:
feels counterintuitive to use `bootloader` to pass an app image via a bootloader flag we should add a `bin` or `app-bin` arg.